service/controller/v4/resource/chart/desired.go

```diff
@@ -101,6 +152,55 @@ func (r *Resource) getSecretValues(ctx context.Context, customObject v1alpha1.Ch
 	return secretValues, nil
 }
 
+// mergeValuesConfigMaps merges values generated by the controlling operator
+// with userValues overriden by the user. Only values under the configmap key
+// can be overriden.
+func mergeValuesConfigMaps(values, userValues map[string]interface{}) (map[string]interface{}, error) {
+	if values == nil || len(values) == 0 {
+		return userValues, nil
+	}
+	if userValues == nil || len(userValues) == 0 {
+		return values, nil
+	}
+
+	// Add any top level user values not present in generated values.
+	for userKey, userVals := range userValues {
+		_, ok := values[userKey]
+		if !ok {
+			values[userKey] = userVals
+		}
+	}
```

Comment:
Is it needed? It's weird that top level values are added only and nested are overriden. What's the use case?

Reply:
Yes we only want users to be able to override settings under configmap. For example we don't want them to be able to change the image that is run.

Reply:
exactly, this is still sth that we are responsible for so we give out only "limited" configurability, otherwise we suddenly need to take care of all kinds of non-standard ICs.

Reply:
Does it make sense to just ignore top level keys?

Reply:
I don't see how that would work. We need to fully merge both maps. There is some complexity here but its to provide a good UX to the user editing the values configmap. The design is written up here.

https://github.com/giantswarm/giantswarm/issues/4185#issue-359542905

Going with this. As it matches the design we're implementing and there is test coverage for the merge logic.